File: Elementary_algo/23_8_3/2.cpp
class Solution {
public:
    int climbStairs(int n) {
        int dp[47] = {0};
        dp[1] = 1;
        for(int i = 2; i <= n + 1; i++)
            dp[i] = dp[i - 1] + dp[i - 2];
        return dp[n + 1];
    }
};

class Solution {
public:
    int climbStairs(int n) {
        int p = 0, q = 0, r = 1;
        for (int i = 0; i < n; i++)
        {
            p = q;
            q = r;
            r = p + q;
        }
        return r;
    }
};

File: Intermediate_algo/linked_list/2.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def oddEvenList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if not head:
            return head
        evenHead = even = head.next
        odd = head
        while even and even.next:
            odd.next = odd = even.next
            even.next = even = odd.next
        odd.next = evenHead
        return head         